### ISSUE TITLE
Some topic references are incorrectly flagged as unresolvable

### ISSUE THREAD

**Issue by natecook1000** [apple/swift-docc#407]
### Description

While updating the curation for the ArgumentParser library, I found that there are some symbols that DocC won't resolve, even though I'm using the symbol names that DocC provided and then displays.

In particular, there are two initializers for the `Argument` type that I used Xcode's DocC integration to autocomplete, but when I build the docs with Xcode _or_ use the `preview-documentation` SwiftPM plugin, I get warnings that the references cannot be found. When I preview the documentation, however, the initializers are actually displayed on the `Argument` type documentation page — not under the correct "Single Arguments" header but in an auto-generated "Initializers" section.

The two topic references are:
- ``init(wrappedvalue:help:completion:)`` ([source](https://github.com/apple/swift-argument-parser/blob/main/Sources/ArgumentParser/Parsable%20Properties/Argument.swift#L308-L312))
- ``init(wrappedvalue:help:completion:transform:)`` ([source](https://github.com/apple/swift-argument-parser/blob/main/Sources/ArgumentParser/Parsable%20Properties/Argument.swift#L376-L381))

You can see the attempted curation here: https://github.com/apple/swift-argument-parser/blob/main/Sources/ArgumentParser/Documentation.docc/Extensions/Argument.md?plain=1#L12-L20

### Checklist

- [x] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/apple/swift-docc/issues).

### Expected Behavior

The initializers are included in the "Single Arguments" section of the `Argument` type documentation page, without any warnings.

### Actual behavior

1. The output of the command includes these messages:
```
/Volumes/Data/Projects/swift-argument-parser/Sources/ArgumentParser/Documentation.docc/Extensions/Argument.md:19:3: warning: Topic reference 'init(wrappedvalue%3Ahelp%3Acompletion%3A)' couldn't be resolved. No local documentation matches this reference.
/Volumes/Data/Projects/swift-argument-parser/Sources/ArgumentParser/Documentation.docc/Extensions/Argument.md:20:3: warning: Topic reference 'init(wrappedvalue%3Ahelp%3Acompletion%3Atransform%3A)' couldn't be resolved. No local documentation matches this reference.
```
2. The two initializers are not included in the "Single Arguments" section of the `Argument` type documentation page.
3. Both initializers are instead shown in the auto-generated "Initializers" section of that page.

### Steps To Reproduce

1. Clone the https://github.com/apple/swift-argument-parser package.
2. Fetch all branches and switch to the `doc-generation` branch.
3. Run the command `swift package --disable-sandbox preview-documentation --target ArgumentParser`

### Swift-DocC Version Information

Noted using the version bundled with Swift 5.7; reproducible on `main`.

### Swift Compiler Version Information

```shell
swift-driver version: 1.62.8 Apple Swift version 5.7 (swiftlang-5.7.0.127.4 clang-1400.0.29.50)
Target: arm64-apple-macosx13.0
```

**Comment by ktoso** [comment]
I've been struggling a lot with this as well, I can provide more examples if necessary. 

It also is incredibly hard to maintain docc references without Xcode, which is an important use case for server developers -- the "could not resolve reference" errors really should list "close matches, otherwise it becomes an incredibly painful dance to pull out Xcode, with apropriate toolchains and generate the docs, then browse the docs and find the right link (since i can't trust Xcode's autocomplete anyway since it results in unresolved links).

**Comment by d-ronnqvist** [comment]
From what I can tell these references are correctly flagged as unresolvable. 

When I build the documentation locally I encounter 6 unresolved links. 5 of those are initializers with a lowercase `v` in `wrappedValue` and the 6th is a disambiguation that doesn't seem to exist for an overload. Because the 5 lowercased `wrappedvalue` links (3 in `Option` and 2 in `Argument`) don't resolve, symbol appear in the automatic "Initializers" sections. This is the same behavior as if the unresolved links didn't exist. The 6th initializer never appear in an automatic "Initializers" section which leads me to think that that symbol doesn't exist anymore. I opened a PR to fix these here https://github.com/apple/swift-argument-parser/pull/511.

With the new link resolver implementation — available behind the `DocCUseHierarchyBasedLinkResolver` feature flag — it's a bit easier to find out how to fix unresolved links but there are still a handful of improvements that I want to make to these error messages.

**Comment by d-ronnqvist** [comment]
> I've been struggling a lot with this as well, I can provide more examples if necessary.

Yes, please provide more examples if you have them since it looks "correct" that the original examples didn't resolve.

> It also is incredibly hard to maintain docc references without Xcode, which is an important use case for server developers -- the "could not resolve reference" errors really should list "close matches", otherwise it becomes an incredibly painful dance to pull out Xcode, with apropriate toolchains and generate the docs, then browse the docs and find the right link (since i can't trust Xcode's autocomplete anyway since it results in unresolved links).

I agree that listing close matches would be a great improvement here. 

Until then, opting in to use the new link resolver implementation should make the workflows outside of Xcode a lot better. If you write a symbol link without any documentation the error message will list all the possible collisions for that symbol name.  

For implementation-detail reasons there are limitations to how link resolution error messages can currently be structured and formatted.

**Comment by d-ronnqvist** [comment]
With the latest bug fixes for the new link resolver in the 5.9 release I believe that this issue is resolved. 

Feel free to reopen this issue if you find more examples of link that are incorrectly flagged as unresolved.
